ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -956,6 +971,16 @@ - (BOOL)shouldPresentStep:(ORKStep *)step {
     return shouldPresent;
 }
 
+- (BOOL)shouldReviewStep:(ORKStep *)step {
+    BOOL shouldReview = ([step isKindOfClass:[ORKQuestionStep class]] || [step isKindOfClass:[ORKFormItem class]]);
+    
+    if (shouldReview && [self.delegate respondsToSelector:@selector(taskViewController:shouldReviewStep:)]) {
+        shouldReview = [self.delegate taskViewController:self shouldReviewStep:step];
```

Comment:
I actually do have a good use case: Use it if you want to filter the steps by some criterion (e.g. only optional steps). One could provide the review step view controller with a segmented control as a custom title view. Based on the selected segment only some of the steps get displayed, the others get filtered out. I think that adding the actual reviewStep is a nice idea for all the delegate methods, not just this one.